### PR TITLE
MAKE-1013: bind as service user before searching

### PR DESCRIPTION
**Pull request by Kimchelly**
JIRA: https://jira.mongodb.org/browse/MAKE-1013

* Add service username and password to config.
* Run isMemberOf (i.e. `userService.search()`) as service user.

### CODE REVIEW

File: ldap/ldap.go [hunk 1]
```diff
@@ -230,8 +244,8 @@ func (u *userService) bind(username, password string) error {
 // search wraps u.conn.Search, reconnecting if the LDAP server has closed the connection.
 // https://github.com/go-ldap/ldap/issues/113
 func (u *userService) search(searchRequest *ldap.SearchRequest) (*ldap.SearchResult, error) {
-	if err := u.ensureConnected(); err != nil {
-		return nil, errors.Wrap(err, "problem connecting to ldap server")
+	if err := u.login(u.serviceUserName, u.serviceUserPassword); err != nil {
```
Comment (bsamek) [inline ldap/ldap.go]: Does this need its own service path as well? I'm not sure if this service user will have the same path as the service users that we currently have.

File: ldap/ldap.go [hunk 2]
```diff
@@ -107,6 +114,13 @@ func (opts CreationOpts) validate() error {
 			opts.URL, opts.Port, opts.UserPath, opts.ServicePath))
 	}
 
+	if opts.ServiceUserName == "" {
+		catcher.Add(errors.New("LDAP service user name cannot be empty"))
+	}
+	if opts.ServiceUserPassword == "" {
+		catcher.Add(errors.New("LDAP service user password cannot be empty"))
```
Comment (tychoish) [inline ldap/ldap.go]: you can do the catcher.NewWhen() to clarify all of these calls

File: ldap/ldap.go [hunk 3]
```diff
@@ -107,6 +114,13 @@ func (opts CreationOpts) validate() error {
 			opts.URL, opts.Port, opts.UserPath, opts.ServicePath))
 	}
 
+	if opts.ServiceUserName == "" {
```
[the remark this answers is not in the document]
Comment (Kimchelly) [reply ldap/ldap.go]: Sure thing. I can't revendor this anyways until we get a service account from IT.

File: ldap/ldap.go (hunk 2 above)
Comment (Kimchelly) [reply ldap/ldap.go]: Done.

File: ldap/ldap.go (hunk 1 above)
Comment (Kimchelly) [reply ldap/ldap.go]: I don't know, but I added one specifically for the service user.

**Pull request merged**